src/overlays/actors/ovl_En_Minislime/z_en_minislime.c

```diff
@@ -70,82 +110,654 @@ static DamageTable D_80985814 = {
+        } else {
+            this->actor.shape.shadowAlpha -= 4;
+        }
+    }
+    Math_StepToF(&this->actor.scale.x, 0.25f, 0.001f);
+    this->actor.scale.z = this->actor.scale.x;
+}
+
+void EnMinislime_SetupDespawn(EnMinislime* this) {
+    this->actionFunc = EnMinislime_Despawn;
+}
+
+void EnMinislime_Despawn(EnMinislime* this, GlobalContext* globalCtx) {
+    Actor_MarkForDeath(&this->actor);
+}
+
+void EnMinislime_SetupMoveToGekko(EnMinislime* this) {
+    this->actor.speedXZ = 0.0f;
+    this->actor.gravity = 0.0f;
+    this->actor.velocity.y = 0.0f;
+    this->collider.base.acFlags &= ~AC_ON;
+    this->collider.base.ocFlags1 &= ~OC1_ON;
+    if ((this->actor.flags & 0x2000) == 0x2000) {
+        this->actor.flags &= ~0x2000;
+    }
+
+    this->actionFunc = EnMinislime_MoveToGekko;
+}
+
+void EnMinislime_MoveToGekko(EnMinislime* this, GlobalContext* globalCtx) {
+    Math_StepToF(&this->actor.world.pos.x, this->actor.parent->world.pos.x, 10.0f);
+    Math_StepToF(&this->actor.world.pos.z, this->actor.parent->world.pos.z, 10.0f);
+    Math_StepToF(&this->actor.world.pos.y, this->actor.parent->world.pos.y + 80.0f, 10.0f);
+    Math_StepToF(&this->actor.scale.x, 0.15f, 0.0075000003f);
+    Math_StepToF(&this->actor.scale.y, 0.075f, 0.0037500001f);
+    Math_StepToF(&this->actor.scale.z, 0.15f, 0.0075000003f);
+    if (this->actor.params == MINISLIME_GEKKO_THROW) {
+        EnMinislime_SetupGekkoThrow(this);
+    } else if (this->actor.params == MINISLIME_IDLE) {
+        EnMinislime_SetupFall(this, globalCtx);
+    }
+}
+
+void EnMinislime_SetupGekkoThrow(EnMinislime* this) {
+    f32 xzDistToPlayer;
+
+    this->collider.base.acFlags |= AC_ON;
+    this->collider.base.ocFlags1 |= OC1_ON;
+    xzDistToPlayer = CLAMP_MIN(this->actor.xzDistToPlayer, 200.0f);
+    this->actor.speedXZ = 17.5f;
+    this->actor.world.rot.y = this->actor.yawTowardsPlayer;
+    this->actor.gravity = -1.0f;
+    this->actor.velocity.y = ((xzDistToPlayer - 200.0f) * 0.01f) + 3.0f;
+    this->actor.bgCheckFlags &= ~1;
+    this->throwTimer = 10;
+    this->actionFunc = EnMinislime_GekkoThrow;
+}
+
+void EnMinislime_GekkoThrow(EnMinislime* this, GlobalContext* globalCtx) {
+    f32 xzScale;
+
+    this->throwTimer--;
+    xzScale = sin_rad(this->throwTimer * (M_PI / 5));
+    this->actor.scale.x = ((0.3f * xzScale) + 1.5f) * 0.1f;
+    this->actor.scale.y = ((cos_rad(this->throwTimer * (M_PI / 5)) * 0.2f) + 0.75f) * 0.1f;
+    this->actor.scale.z = 0.3f - this->actor.scale.x;
+    if (this->throwTimer == 0) {
+        this->throwTimer = 10;
+    }
+
+    if (this->actor.bgCheckFlags & 1) {
+        EnMinislime_SetupGrowAndShrink(this);
+    }
+}
+
+void EnMinislime_ApplyDamage(EnMinislime* this) {
+    if (this->collider.base.acFlags & AC_HIT) {
+        this->collider.base.acFlags &= ~AC_HIT;
+        if (this->actionFunc == EnMinislime_IceArrowDamage) {
+            if (this->actor.colChkInfo.damageEffect != 14) {     // Hookshot
```

Comment:
So working with Sakon, I learned that these effect values (i.e 14) are actually the second parameter of the DMG_ENTRY macro. Since those are in Hex, I feel these should also be in hex. Though this might need a bigger discussion for consistency.

Reply:
Yeah perhaps it's worth discussing, but I think I like hex a bit better in the context of the DMG_ENTRY macros that are in

Reply:
What should actually happen with documentation is to make an enum that describes the effect (or the cause of the effect) (see e.g. EnFirefly or EnBili or EnBari in OoT). We couldn't do this without damage table macros, but we can now.

Reply:
I'll probably add that enum then for both Minislime and Pametfrog